lib/beaker/hypervisor/openstack.rb

```diff
@@ -115,13 +115,27 @@ def provision
         # Associate a public IP to the server
         # Create if there are no floating ips available
         #
-        ip = @compute_client.addresses.find { |ip| ip.instance_id.nil? }
-        if ip.nil?
-          @logger.debug "Creating IP for #{host.name} (#{host[:vmhostname]})"
-          ip = @compute_client.addresses.create
+        # Do we already have an address?
+        @logger.debug vm.addresses
+
+        begin
+          if vm.addresses[@options[:openstack_network]]
+            address = vm.addresses[@options[:openstack_network]].map{ |network| network['addr'] }.first
+          end
+        rescue
```

Comment:
Please determine a reasonable set of catchable exceptions - we do not support empty rescue statements, especially with no error messaging.  It just makes bug tracking impossible in the long run.